models/task.go

```diff
@@ -22,6 +26,24 @@ type TaskLinks struct {
 	Job  string `json:"job"`
 }
 
+// TaskName - iota enum of possible task names
+type TaskName int
+
+// ParseTaskName returns a TaskName from its string representation
+func ParseTaskName(taskNameStr string) (TaskName, error) {
```

Comment:
I don't see the need for this function returning `TaskName` as it is thrown away by the calling method on line 76 inside Validate method.